docs/editor/intellisense.md

```diff
@@ -0,0 +1,156 @@
+
+You can trigger IntelliSense in any editor window by typing `kb(editor.action.triggerSuggest)` or by typing a trigger character (such as `.` in JavaScript). 
+
+![intellisense in package json](images/intellisense/intellisense_packagejson.gif)
+
+> **Tip:** The suggestions filtering supports CamelCase so you can type the upper case letters of a method name to limit the suggestions. For example, "cra" will quickly bring up "createApplication".
+
+> **Tip:** You can turn off IntelliSense while you type. See [Customize IntelliSense below](#_customize_intellisense).
+
+In addition, you can see **quick info** for each method (as provided by the language service). 
+
+![quick info](images/intellisense/quick_outline.png)
+
+After choosing a method you are provided with **parameter info**. 
+
+![paramater info](images/intellisense/paramater_info.png)
+
+
+## Suggestion Priority
+
+The order and the icon to the left of the word show the IntelliSense suggestion's priority.  
+
+> **Higher suggestion priority = higher order and a more detailed icon**
+
+This image shows the result of triggering IntelliSense on `express` a popular [Node.js module](https://expressjs.com/). 
+
+![image showing intellisense icons](images/intellisense/intellisense_icons.png)
+
+Methods, variables, and objects tend to be high priority. Simple word completion, tend to be lower priority. When the priority level is equivalent, ordering is done alphabetically. 
+
+|       |         |
+| ----- | ------- |
+| ![purple cube for a method](images/intellisense/method_icon.png) | Method or Function |
+| ![blue cuboid for a variable](images/intellisense/variable_icon.png) | Variable | 
+| ![blue circles connected by a blue line for an object](images/intellisense/object_icon.png) | Object |
+| ![a square with a small fold in the top left corner indicates word completion](images/intellisense/word_completion_icon.png) | Word Completion |
+| ![a square with a small fold in the top left corner indicates string completion](images/intellisense/string_completion_icon.png) | String completion |
+
+## Customize IntelliSense
+
+You can customize your IntelliSense experience in settings and key bindings. 
+
+### Settings
+
+The settings shown below are the default settings. You can change these settings in your settings.json file as described [here](/docs/customization/userandworkspace.md). 
+
+```json
+{
+
+    // Controls if quick suggestions should show up while typing
+    "editor.quickSuggestions": true,
+
+    // Controls if suggestions should be accepted with "Enter" - in addition to "Tab". Helps to avoid ambiguity between inserting new lines and accepting suggestions. 
+    "editor.acceptSuggestionOnEnter": true,
+
+    // Controls the delay in ms after which quick suggestions will show up. 
+    "editor.quickSuggestionsDelay": 10
+}
+```
+
```

Comment:
Perhaps switch this block to **js**. Comments don't really exist in json so GitHub butchers them in code blocks. It might not be accurate, but it looks better.
```js  
{    
    // Controls if quick suggestions should show up while typing  
    "editor.quickSuggestions": true,  
    // Controls if suggestions should be accepted with "Enter" - in addition to "Tab". Helps to avoid ambiguity  between inserting new lines and accepting suggestions.   
    "editor.acceptSuggestionOnEnter": true,  
     // Controls the delay in ms after which quick suggestions will show up.   
    "editor.quickSuggestionsDelay": 10  
}  
```